charts/shoot-core/components/charts/nodelocal-dns/templates/daemonset.yaml

```
@@ -0,0 +1,102 @@
+---
+apiVersion: {{ include "daemonsetversion" . }}
+kind: DaemonSet
+metadata:
+  name: node-local-dns
+  namespace: kube-system
+  labels:
+    gardener.cloud/role: system-component
+    garden.sapcloud.io/role: system-component
+    origin: gardener
+    k8s-app: node-local-dns
+spec:
+  updateStrategy:
+    rollingUpdate:
+      maxUnavailable: 10%
+  selector:
+    matchLabels:
+      k8s-app: node-local-dns
+  template:
+    metadata:
+      labels:
+        k8s-app: node-local-dns
+        networking.gardener.cloud/to-dns: allowed
+        # we won't be using the checksum of the configmap since node-local-dns provides the "reload" plugins that does the reload if config changes.
+      annotations:
+        prometheus.io/port: "{{ .Values.prometheus.port }}"
+        prometheus.io/scrape: "{{ .Values.prometheus.scrape }}"
+    spec:
+      priorityClassName: system-node-critical
+      serviceAccountName: node-local-dns
+      hostNetwork: true
+      dnsPolicy: Default  # Don't use cluster DNS.
+      tolerations:
+        - key: CriticalAddonsOnly
+          operator: Exists
+        - effect: NoExecute
+          operator: Exists
+        - effect: NoSchedule
+          operator: Exists
+      containers:
+        - name: node-cache
+          image: {{ index .Values.images "node-local-dns" }}
+          resources:
+            requests:
+              cpu: 25m
+              memory: 5Mi
+            limits:
+              cpu: 100m
+              memory: 25Mi
```

```suggestion
              memory: 25Mi
            limits:
              cpu: 100m
              memory: 100Mi
```

ref https://github.com/gardener/gardener/pull/2164